src/systems/camera.js

```diff
@@ -57,30 +69,58 @@ module.exports.System = registerSystem('camera', {
    * will return the real user's head height and position.
    */
   addDefaultOffset: function () {
-    var defaultCamera = this.sceneEl.querySelector('[' + DEFAULT_CAMERA_ATTR + ']');
+    var cameraOffset;
     var currentPosition;
-    if (!defaultCamera) { return; }
+    var defaultCamera = this.sceneEl.querySelector('[' + DEFAULT_CAMERA_ATTR + ']');
+    // Only applies if there's a default camera with no applied offset.
+    if (!defaultCamera || this.cameraOffset) { return; }
     currentPosition = defaultCamera.getAttribute('position');
+    // The polyfill introduces an offset in the y axis for human scale perspective.
+    if (isMobile) { currentPosition.y = 0; }
+    cameraOffset = this.cameraOffset = DEFAULT_CAMERA_POSITION;
     defaultCamera.setAttribute('position', {
-      x: currentPosition.x + DEFAULT_CAMERA_POSITION.x,
-      y: currentPosition.y + DEFAULT_CAMERA_POSITION.y,
-      z: currentPosition.z + DEFAULT_CAMERA_POSITION.z
+      x: currentPosition.x + cameraOffset.x,
+      y: currentPosition.y + cameraOffset.y,
+      z: currentPosition.z + cameraOffset.z
     });
   },
 
   removeDefaultOffset: function () {
     // Remove default camera if present.
-    var defaultCamera = this.sceneEl.querySelector('[' + DEFAULT_CAMERA_ATTR + ']');
+    var cameraOffset = this.cameraOffset;
     var currentPosition;
-    if (!defaultCamera) { return; }
+    var sceneEl = this.sceneEl;
+    var defaultCamera = sceneEl.querySelector('[' + DEFAULT_CAMERA_ATTR + ']');
+    // Checking this.headsetConnected to make the value injectable for unit tests.
+    var headsetConnected = this.headsetConnected || checkHeadsetConnected();
```

Comment:
`this.headsetConnected` check doesn't seem needed.

If so then we can call `if (!defaultCamera || !cameraOffset || !checkHeadsetConnected()) { return; }` directly